src/services/EcoBridge/Socket/Socket.reducer.ts

```diff
@@ -0,0 +1,141 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { ChainId } from '@swapr/sdk'
+import { SocketBridgeState, SocketTx, SocketTxStatus } from './Socket.types'
+import { TokenList } from '@uniswap/token-lists'
+import { SocketList, SyncState, BridgingDetailsErrorMessage, BridgeDetails } from '../EcoBridge.types'
+import { Route } from './api/generated'
+
+const initialState: SocketBridgeState = {
+  transactions: [],
+  approvalData: {},
+  txBridgingData: {},
+  bridgingDetails: {},
+  bridgingDetailsStatus: SyncState.IDLE,
+  listsStatus: SyncState.IDLE,
+  lists: {},
+  routes: [],
+  lastMetadataCt: 0
+}
+
+const createSocketSlice = (bridgeId: SocketList) =>
+  createSlice({
+    name: bridgeId,
+    initialState,
+    reducers: {
+      setBridgeDetails: (state, action: PayloadAction<BridgeDetails>) => {
+        const { gas, fee, estimateTime, receiveAmount, requestId } = action.payload
+
+        //(store persist) crashing page without that code
+        if (!state.bridgingDetails) {
+          state.bridgingDetails = {}
+        }
+
+        if (requestId !== state.lastMetadataCt) {
+          if (state.bridgingDetailsStatus === SyncState.FAILED) return
+          state.bridgingDetailsStatus = SyncState.LOADING
+          return
+        } else {
+          state.bridgingDetailsStatus = SyncState.READY
+        }
+
+        if (gas) {
```

Comment:
I made a similar comment elsewhere. Do we want to use if/elses instead of just sequence of ifs?

Reply:
We would rather leave it that way.